docker/Dockerfile

```
@@ -1,55 +1,66 @@
-FROM ubuntu:16.04 
+FROM openjdk:11.0.8-jre-slim-buster
 ENV ZK_USER=zookeeper \
 ZK_DATA_DIR=/var/lib/zookeeper/data \
 ZK_DATA_LOG_DIR=/var/lib/zookeeper/log \
 ZK_LOG_DIR=/var/log/zookeeper \
-JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
+ZK_DOWNLOAD_URL=https://downloads.apache.org/zookeeper \
+APACHE_KEYS="68E327C1 6E1CC7A4 8EE2F25C D2C80E32 820F225C 92BC2F2B ADAFD097 \
+  B5E9BD2D 767E7473 41CF31A9 173C31A2 15DFA1BA A615D22C 3B5B5147 8F6E35FC"
 
-ARG GPG_KEY=C823E3E5B12AF29C67F81976F5CECB3CB5E9BD2D
-ARG ZK_DIST=zookeeper-3.4.10
 RUN set -x \
     && apt-get update \
-    && apt-get install -y openjdk-8-jre-headless wget netcat-openbsd \
-	&& wget -q "http://www.apache.org/dist/zookeeper/$ZK_DIST/$ZK_DIST.tar.gz" \
-    && wget -q "http://www.apache.org/dist/zookeeper/$ZK_DIST/$ZK_DIST.tar.gz.asc" \
-    && export GNUPGHOME="$(mktemp -d)" \
-    && gpg --keyserver ha.pool.sks-keyservers.net --recv-key "$GPG_KEY" \
-    && gpg --batch --verify "$ZK_DIST.tar.gz.asc" "$ZK_DIST.tar.gz" \
-    && tar -xzf "$ZK_DIST.tar.gz" -C /opt \
-    && rm -r "$GNUPGHOME" "$ZK_DIST.tar.gz" "$ZK_DIST.tar.gz.asc" \
-    && ln -s /opt/$ZK_DIST /opt/zookeeper \
+    && apt-get -qq -y --no-install-recommends install \
+      gnupg \
+      wget \
+      netcat-openbsd \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN for key in $APACHE_KEYS; do \
+      gpg --keyserver ha.pool.sks-keyservers.net --recv-key "$key" || \
+      gpg --keyserver keyserver.ubuntu.com --recv-key "$key"; \
+    done
+
+ARG ZK_DIST=zookeeper-3.4.14
+ARG ZK_PACKAGE=$ZK_DIST
+RUN wget -q "$ZK_DOWNLOAD_URL/$ZK_DIST/$ZK_PACKAGE.tar.gz" \
+    && wget -q "$ZK_DOWNLOAD_URL/$ZK_DIST/$ZK_PACKAGE.tar.gz.asc"
+
+RUN gpg --batch --verify "$ZK_PACKAGE.tar.gz.asc" "$ZK_PACKAGE.tar.gz" \
```

It's better to do all zookeeper installation related stuff on same layer in order to let image smaller
```suggestion
    && wget -q "$ZK_DOWNLOAD_URL/$ZK_DIST/$ZK_PACKAGE.tar.gz.asc" \
    && gpg --batch --verify "$ZK_PACKAGE.tar.gz.asc" "$ZK_PACKAGE.tar.gz" \
```